src/application/views/WelcomeView.tsx

```diff
@@ -1,30 +1,60 @@
-import { useQuery } from '@tanstack/react-query';
-import { Fragment } from 'react';
+import { useAtomValue } from 'jotai';
 
-import { ProjectInfo, readAllProjects } from '../../api/projectsAPI';
+import { allProjectsAtom } from '../../atoms/projectState';
 import { Button } from '../../components/Button';
 import { OpenIcon } from '../../components/icons';
 import { emitEvent } from '../../events';
 import { cx } from '../../lib/cx';
-import { AddIcon, RefStudioEditorIcon, SampleIcon } from '../components/icons';
+import { AddIcon, EmptyStateIcon, SampleIcon } from '../components/icons';
+import { ProjectsList } from '../components/ProjectsList';
 
 export function WelcomeView() {
+  const projects = useAtomValue(allProjectsAtom);
+
```

Comment:
How is the list of projects (atom) updated?
I think we should use an effect to update the atom one we land on this view. What do you think?

Reply:
You're right! Good catch!
I'm wondering whether the update should be done when creating a project instead? The issue with updating it here is that we would first render the outdated list, then new projects would appear once the list is updated and that could look weird because there would not be any loading effect

Reply:
I've also thought about that. My concern was that the list can be updated externally. For now your idea is a good solution.